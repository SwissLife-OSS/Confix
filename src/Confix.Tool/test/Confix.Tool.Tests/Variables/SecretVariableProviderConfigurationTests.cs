using System.Text.Json.Nodes;
using ConfiX.Variables;
using FluentAssertions;

namespace Confix.Tool.Tests;

public class SecretVariableProviderConfigurationTests
{

    [Fact]
    public void Parse_EmptyObject_DefaultAlgorithm()
    {
        // arrange
        var configuration = JsonNode.Parse("{}")!;

        // act
        var result = SecretVariableProviderConfiguration.Parse(configuration);

        // assert
        result.Should().Be(new SecretVariableProviderConfiguration
        {
            Algorithm = SecretVariableProviderAlgorithm.RSA,
        });
    }

    [Fact]
    public void Parse_WithValidConfigurationKey_ReturnsValidObject()
    {
        // arrange
        var configuration = JsonNode.Parse(
            """
            {
                "algorithm": "RSA",
                "publicKey": "SomePublicKey",
                "privateKey": "SomePrivateKey"
            }
            """
        )!;

        // act
        var result = SecretVariableProviderConfiguration.Parse(configuration);

        // assert
        result.Should().Be(new SecretVariableProviderConfiguration
        {
            Algorithm = SecretVariableProviderAlgorithm.RSA,
            PublicKey = "SomePublicKey",
            PrivateKey = "SomePrivateKey",
        });
    }

    [Fact]
    public void Parse_WithValidConfigurationPath_ReturnsValidObject()
    {
        // arrange
        var configuration = JsonNode.Parse(
            """
            {
                "algorithm": "RSA",
                "publicKeyPath": "./pub.pem",
                "privateKeyPath": "./priv.pem"
            }
            """
        )!;

        // act
        var result = SecretVariableProviderConfiguration.Parse(configuration);

        // assert
        result.Should().Be(new SecretVariableProviderConfiguration
        {
            Algorithm = SecretVariableProviderAlgorithm.RSA,
            PublicKeyPath = "./pub.pem",
            PrivateKeyPath = "./priv.pem",
        });
    }
}
