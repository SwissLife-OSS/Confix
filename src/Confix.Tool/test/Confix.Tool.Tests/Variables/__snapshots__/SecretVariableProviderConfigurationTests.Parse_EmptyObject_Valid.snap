{
  "Algorithm": null,
  "Padding": null,
  "PublicKey": null,
  "PublicKeyPath": null,
  "PrivateKey": null,
  "PrivateKeyPath": null,
  "Password": null
}
